pkg/config/config.go

```diff
@@ -568,6 +568,15 @@ type ImageConfig struct {
 	// containing credentials necessary for pulling images from secure
 	// registries.
 	GlobalAuthFile string `toml:"global_auth_file"`
+	// NamespacedAuthDir is the root path for pod namespace-separated
+	// auth files. The final auth file will be <NAMESPACED_AUTH_DIR>/<NAMESPACE>-<IMAGE_NAME_SHA256.json.
+	// The final auth values will be used instead of any configured GlobalAuthFile.
+	// The auth file will be automatically removed by CRI-O after its usage.
+	// If no pod namespace is being provided on image pull (via the sandbox
+	// config), or the concatenated path is non existent, then the
+	// system wide auth file will be used as fallback.
```

Comment:
Is this feature for extending the way of authentication, or for restricting access in specific namespaces?
I'm curious whether users want it to fall back to global auth or just fail, when there's no auth file for the image though there are some auth files for other images in the same namespace.